File: Programmers/src/Javascript/유한소수판별하기.js
function solution(a, b) {
    let under = b / getGCD(a,b);
    
    while(under != 1) {
        if(under%2 === 0)
            under /= 2;
        else if(under%5 === 0)
            under /= 5;
        else
            return 2;
    }
    
    return 1;
}

let getGCD = (num1, num2) => {
    if(num2 === 0)
        return num1;
    else
        return getGCD(num2, num1%num2);
}

File: Programmers/src/Java/Lv0/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        int B = b / gcd(a,b);
        
        while(B != 1) {
            if(B%2==0)
                B/=2;
            else if(B%5==0)
                B/=5;
            else
                return 2;
        }
        
        return 1;
    }
    
    public static int gcd(int a, int b) {
        if(b==0)
            return a;
        else
            return gcd(b, a%b);
    }
}